### PR TITLE
Fixing publishZip publication by adding the dependency to generatePomFileForNebulaPublication

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Fixing `publishZip` publication by adding the dependency to `generatePomFileForNebulaPublication`, the issue manifested in https://github.com/opensearch-project/opensearch-plugin-template-java/pull/31:

```
> Task :publishPluginZipPublicationToZipStagingRepository
Execution optimizations have been disabled for task ':publishPluginZipPublicationToZipStagingRepository' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: 'opensearch-plugin-template-java/build/distributions/rename-unspecified.pom'. Reason: Task ':publishPluginZipPublicationToZipStagingRepository' uses this output of task ':generatePomFileForNebulaPublication' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.4.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
```
 
### Issues Resolved
N/A
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@prudhvigodithi could you please take a look? (address one of the issues).

**Comment by prudhvigodithi** [comment]
Hey @reta, can you please specify, how did you invoke the task, for me `publishPluginZipPublicationToZipStagingRepository` say SUCCESSFUL, but what throws warning is `./gradlew  build`, are we on the same page ?
```[opensearch@11827a526d02 opensearch-plugin-template-java]$ ./gradlew publishPluginZipPublicationToZipStagingRepository
=======================================
OpenSearch Build Hamster says Hello!
  Gradle Version        : 7.4.2
  OS Info               : Linux 5.4.186-113.361.amzn2int.x86_64 (amd64)
  JDK Version           : 14 (AdoptOpenJDK)
  JAVA_HOME             : /usr/lib/jvm/adoptopenjdk-14-hotspot
  Random Testing Seed   : AC054586B1C8F365
  In FIPS 140 mode      : false
=======================================

BUILD SUCCESSFUL in 1s
```

**Comment by reta** [comment]
> Hey @reta, can you please specify, how did you invoke the task, for me `publishPluginZipPublicationToZipStagingRepository` say SUCCESSFUL, but what throws warning is `./gradlew build`, are we on the same page ?

```
./gradlew clean build publishPluginZipPublicationToZipStagingRepository 
```

**Comment by prudhvigodithi** [comment]
@reta thanks for clue, can you please check/test the PR I have made it little better.
https://github.com/opensearch-project/OpenSearch/pull/3189/files

**Review by dblock — APPROVED** [review]
Any reason not to merge this @prudhvigodithi ?

**Comment by prudhvigodithi** [comment]
Hey @dblock, I have created an another PR that fixes both the tasks dependency issues
PR : https://github.com/opensearch-project/OpenSearch/pull/3189
@reta added some feedback there and once we merge the other [PR](https://github.com/opensearch-project/OpenSearch/pull/3189), we need not merge this PR as above PR address the Fix @reta

**Comment by dblock** [comment]
Closing in favor of https://github.com/opensearch-project/OpenSearch/pull/3189

**Pull request closed without merging**
